res/controllers/engine-api.d.ts

```diff
@@ -25,6 +25,7 @@ declare interface ScriptConnection {
 
 declare namespace engine {
     type SettingValue = string | number | boolean;
+    type SharedDataValue = string | number | boolean | object | array | undefined;
```

Comment:
```suggestion
    type SharedDataValue = any;
```
at that point there is no need for the alias imo. It only worsens the editor UX

Reply:
Why? Can you give me a specific example where it worsen the experience?

Reply:
1. Do we want to share complex data types or only basic data types?
2. If we allow any data type, than we should use the keyword `any` instead of an alias.

Reply:
Well, trying this API out in VSCode, this is the intellisense I get: 
![image](https://github.com/mixxxdj/mixxx/assets/12380386/f11f97b6-0bef-4e6f-80b1-ac7ba0e674b1)
You can see how the type alias is opaque. 

Interestingly this is only the case if I reduce the number of types `SharedDataValue` could be. If I leave it with the current set, its automatically coerced to `any`...

![image](https://github.com/mixxxdj/mixxx/assets/12380386/be7a4448-3edf-430f-96b1-096452938193)

At that point, I'd say we can replace the entire set with `any`. For some reason the type alias then becomes opaque. So I guess we can keep it if you insist.

Reply:
> Do we want to share complex data types or only basic data types?

Yes, I think we do. For example, one may want to share a `Uint8Array`, used as a HID packet, so it can be send to two or more devices

> If we allow any data type, than we should use the keyword any instead of an alias.

We don't - for example `callable` is not meant to be shared.
The reason why I disagree with using `any` is that it goes against TypeScript. Now if you think it just makes it simpler, I'm not going to argue.

Reply:
> Yes, I think we do. For example, one may want to share a `Uint8Array`, used as a HID packet, so it can be send to two or more devices

I don't see a use case for sharing such low level data between scripts. We should really limit this to high level syncronisation like a deck-switch or a global shift button. For everything else, we have existing APIs available.

> 
> > If we allow any data type, than we should use the keyword any instead of an alias.
> 
> We don't - for example `callable` is not meant to be shared. The reason why I disagree with using `any` is that it goes against TypeScript. Now if you think it just makes it simpler, I'm not going to argue.

If this is not equal `any`, than we should keep the type definition!

Reply:
> I don't see a use case for sharing such low level data between scripts. We should really limit this to high level syncronisation like a deck-switch or a global shift button. For everything else, we have existing APIs available.

I disagree. You should be able to share any memory representation you want, whether it is `class MySharedContext`, or a `TypedScript`. Since we are now focusing on future problems that could arise with this solution, we should also consider slightly unorthodox case and not introduce limitations IMO.

Reply:
> The reason why I disagree with using any is that it goes against TypeScript.

I think the issue is that technically according to the signature, sharing a callable would be possible because a callable is also an object, which is allowed. There is probably a reason why vscode coerced the current signature to `any` automatically. but take my words with a grain of salt, I don't really know typescript.